xcodeproj/internal/xcodeprojinfo.bzl

```diff
@@ -259,6 +260,15 @@ def _skip_target(
         if attr in deps_attrs and info.xcode_target
     ]
 
+    test_suite_replacement_labels = []
+    if ctx.rule.kind == "test_suite":
+        for info in valid_transitive_infos:
+            for xcode_target in info.xcode_targets.to_list():
+                if xcode_target.id.count(".__internal__.__test_bundle"):
+                    test_suite_replacement_labels.append(
+                        struct(id = xcode_target.id, label = target.label),
+                    )
+
```

Comment:
@brentleyjones curious if you see a better approach here. Without this the scheme is Xcode gets named after the latest label from the underlying runners: https://github.com/bazel-ios/rules_ios/blob/master/tests/macos/xcodeproj/BUILD.bazel#L55 (which was the `iPhone` one based on my testing)

But when using a test suite I'd say the expected behaviour is to see the name of the respective test suite in Xcode: https://github.com/bazel-ios/rules_ios/blob/master/tests/macos/xcodeproj/BUILD.bazel#LL41C13-L41C49

All replacements are happening on top of the same label for the internal `.__internal__.__test_bundle` target, so adding this as the last option below as `+ test_suite_replacement_labels` fixed it for me but maybe there's a better approach here

Also this fixed adding the `test_suite` rule as a custom scheme in `schemes`, otherwise if fails since it can't find the `test_suite` label in the transitive deps and finds only the `iPhone/iPad` ones, so It ends up hitting this: https://github.com/MobileNativeFoundation/rules_xcodeproj/blob/main/tools/generator/src/Extensions/XcodeScheme%2BExtensions.swift#L27

Reply:
When it's a test suite, I think we shouldn't set `depset(direct)` below in `replacement_labels`, since we want the labels to be left alone, right?

Reply:
>When it's a test suite, I think we shouldn't set depset(direct) below in replacement_labels, since we want the labels to be left alone, right?

Ok, let me look into that!

>I don't think we want the test suite to appear in Xcode, I think the various sub-tests should just be included.

Maybe I'm thinking about this backwards, I thought it made sense to have the test suite there and Xcode is the "runner" but now I think that missed the point of having different runners already configured in `BUILD` files. Ok, let me explore the approach above.

Reply:
> I don't think we want the test suite to appear in Xcode, I think the various sub-tests should just be included.

It could become an auto-generated scheme that builds & tests all the tests under it maybe? But right, I think the default would be that it simply includes whatever test targets are declared in the suite.

Reply:
Yep, having all underlying tests in Xcode seems to be the correct behaviour indeed, discussing possible approaches offline to not pollute this thread too much but the TL;DR is that both underlying tests depend on the same `.__internal__.__test_bundle` test bundle target which is causing the latest one in `replacement_labels` to override all the others and we end up with a single test target in Xcode

Reply:
>It could become an auto-generated scheme that builds & tests all the tests under it maybe

This seems wroth exploring, haven't looked into auto-generated schemes at all but will do

Reply:
Ok, after some discussion I'll remove this and go with a different approach, will detailed it better in the PR descr once it's ready for review